lib/runtime/console2.js

```diff
@@ -257,8 +252,14 @@ function addLinkHandler (terminal) {
   terminal.registerLinkMatcher(uriRegex, handleLink)
 }
 
-function handleWhitelistedKeybindingTerminal (e) {
-  if (whitelistedKeybindingsTerminal.indexOf(atom.keymaps.keystrokeForKeyboardEvent(e)) > -1) {
+function handleKeybinding (e, term, binds = whitelistedKeybindingsTerminal) {
+  // Meta-Enter doesn't work properly with xterm.js atm, so we send the right escape sequence ourselves:
+  if (e.keyCode === 13 && (e.altKey || e.metaKey)) {
+    if (term.ty) {
+      term.ty.write('\x1b\x0d')
+    }
+    return false
+  } else if (binds.indexOf(atom.keymaps.keystrokeForKeyboardEvent(e)) > -1) {
     return false
   }
   return e
```

Comment:
As you have it currently, the comment looks like it's describing the entire `if-elseif-end` block. Maybe you should put the comment inside the if? And/or also add a comment on the `else if` line?

Reply:
Fair point. Fixed.